client/engine/engine.go

```diff
@@ -238,15 +238,15 @@ func (e *Engine) constructObjectiveFromMessage(message protocols.Message) (proto
 	case strings.Contains(string(message.ObjectiveId), `DirectFund`):
 		initialState := message.SignedStates[0].State()
 		if initialState.TurnNum != 0 {
-			return directfund.DirectFundObjective{}, errors.New("cannot construct direct fund objective without prefund state")
+			return directfund.Objective{}, errors.New("cannot construct direct fund objective without prefund state")
 		}
-		return directfund.New(
+		return directfund.NewObjective(
 			true, // TODO ensure objective in only approved if the application has given permission somehow
 			message.SignedStates[0].State(),
 			*e.store.GetAddress(),
 		)
 	default:
-		return directfund.DirectFundObjective{}, errors.New("cannot handle unimplemented objective type")
+		return directfund.Objective{}, errors.New("cannot handle unimplemented objective type")
```

Comment:
... and here's the other part of the motivation. As is, we end up repeating ourselves unnecessarily.